src/main/scala/scalismo/statisticalmodel/asm/Profiles.scala

```diff
@@ -17,9 +18,31 @@ case class Profiles(domain: SpatiallyIndexedDiscreteDomain[_3D], data: immutable
   override def values = data.iterator
 
   override def interpolateNearestNeighbor(): Field[_3D, MultivariateNormalDistribution] = {
-    val indexedDomain = SpatiallyIndexedDiscreteDomain(domain.points.toIndexedSeq, domain.numberOfPoints)
-    Field(domain.boundingBox, (p: Point[_3D]) => apply(indexedDomain.findClosestPoint(p)._2))
+    Field(domain.boundingBox, (p: Point[_3D]) => apply(domain.findClosestPoint(p)._2))
   }
 
 }
 
+/**
+ * Class of Discrete Fields where to each point, a DenseVector of arbitrary dimensionality is associated.
+ *
+ * An example instance of such a class is the set of profile points and associated features in an Active Shape Model.
+ *
+ */
+
+case class DiscreteFeatureField[D <: Dim: NDSpace](private val pointsAndValues: IndexedSeq[(Point[D], DenseVector[Float])]) extends DiscreteField[D, DenseVector[Float]] {
```

Comment:
Should this stay in the asm subpackage, or should it go into the "common" package where the other fields are defined?